src/main/java/inventorysetups/InventorySetupsPersistentDataManager.java

```diff
@@ -210,26 +272,42 @@ private void handleMigrationOfOldData()
 		if (Strings.isNullOrEmpty(hasMigratedToV2))
 		{
 			log.info("Migrating data to V2");
-			// Perform migration of old data
-			Type setupType = new TypeToken<ArrayList<InventorySetup>>()
+			inventorySetups.addAll(loadV1Setups());
+			updateConfig(true, false);
+			inventorySetups.clear();
+			configManager.setConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2, "True");
+		}
+
+		hasMigratedToV2 = configManager.getConfiguration(CONFIG_GROUP, CONFIG_KEY_SETUPS_MIGRATED_V2);

```

Comment:
I think the wrapping if statement and `hasMigratedToV2` assignment can be removed. Based on the order in the function, the data is already migrated when we reach here. So there is no need to check.

Reply:
done